### PR TITLE
Updates for Mix

### PR DESCRIPTION
**Pull request by srtfisher**
Add parts of #117 to improve the Mix integration with Mantle. Sets up for automatic entries and block enqueuing.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,21 +1,122 @@
...
+    "prop-types": "^15.8.0",
+    "react": "17.0.2",
+    "react-dom": "17.0.2",
+    "styled-components": "^5.3.1",
+    "uuid": "^8.3.2"
+  },
+  "devDependencies": {
+    "@types/jest": "^29.0.3",
+    "@babel/preset-react": "^7.18.6",
+    "@tinypixelco/laravel-mix-wp-blocks": "^1.2.0",
```
Comment (anubisthejackle) [inline package.json:110]: This should handle all of the @wordpress packages for us, no?

Comment (srtfisher) [reply package.json:110]: What I struggle with is IDE completion. Without these packages, I can't get it unless I use a `@types/` or something. Will dig in.

File: jsconfig.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "compilerOptions": {
+    "baseUrl": ".",
+    "paths": {
+      "@/*": ["*"]
+    }
+  }
+}
```
Comment (stevenslack) [inline jsconfig.json:8]: If Typescript is supported this file could be eliminated in favor of the `tsconfig.json`

File: package.json [hunk 3]
```diff
@@ -1,21 +1,122 @@
...
+    "hot": "mix watch --hot",
+    "preprod": "check-node-version --package",
+    "prod": "npm run production",
+    "production": "mix --production",
+    "pretest:watch": "check-node-version --package",
+    "test:watch": "jest --watch",
+    "pretest": "check-node-version --package",
+    "test": "jest",
+    "preupdate-dependencies": "check-node-version --package",
+    "update-dependencies": "node bin/update-dependencies.js",
```
Comment (stevenslack) [inline package.json:21]: If the update dependencies script is swapped in favor or `@wordpress/scripts` you could add the script here. Example in the starter plugin: https://github.com/alleyinteractive/wp-starter-plugin/blob/feature/wp-scripts/package.json#L33-L34

File: src/webpack/discover-entries.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  /**
+   * Discover entries in the specified directory.
+   */
+  discoverEntries() {
+    const entries = glob
+      .sync(`${rootPath}/${this.entryPath}/**/index.?(j|t)s*`)
+      .reduce((acc, item) => {
+        const entry = item
+          .replace(`${rootPath}/${this.entryPath}/`, '')
+          .replace(/\/index\.jsx?/, '');
```
Comment (stevenslack) [inline src/webpack/discover-entries.js:53]: Maybe supply a regex for `.tsx` as well?

File: webpack.mix.js [hunk 5]
```diff
@@ -10,20 +14,90 @@ require('laravel-mix-polyfill');
...
+| index.js files that live within a folder inside entries will automatically be
+| compiled.
+|
+| Discover index.js files within a specific directory:
+|
+|   mix.discover('entries');
+|
+| Discover index.ts files within a specific directory using Typescript:
+|
+|  mix.discover('entries', { type: 'ts' });
```
Comment (stevenslack) [inline webpack.mix.js:48]: Could it not detect `ts` files as well as `js` files without specification? Looks like you have the logic here: https://github.com/alleyinteractive/mantle/pull/118/files#diff-0da3eaa1f1380aee7113ee4a1c26d26b1e425f1b03994413a7fdf5b068b654bfR49

I am not that familiar with Mix, so I could be missing something.

File: webpack.mix.js [hunk 6]
```diff
@@ -10,20 +14,90 @@ require('laravel-mix-polyfill');
...
+|
+| Discover WordPress blocks within /blocks/:
+|
+|   mix.discover('blocks', { type: 'block' });
+*/
+
+mix.discover('entries');
+mix.discover('blocks', { type: 'block' });
```
Comment (stevenslack) [inline webpack.mix.js:56]: I would love to discuss compiling blocks with `block.json` involved!

**Review by stevenslack — APPROVED** [review]
Nothing blocking, but added some comments for consideration. 🐞

**Comment by srtfisher** [comment]
@stevenslack thank you for all that feedback, that really helped. I've switched to use `.ts` files by default and included all the relevant packages. I also switched to using `@wordpress/scripts`, too. Really neat find!

I'd love to regroup on this and the starter plugins to solidify shared initiates -- this one is almost ready. @anubisthejackle is wrapping up Gutenberg block generation and then we'll write up some documentation on it that make it very easy to generate blocks with Mantle.

**Pull request merged**